docs/develop/best-practices.md

```diff
@@ -0,0 +1,164 @@
+  const TargetTableName = 'Table1';
+  const SourceTableName = 'Table2';
+
+  // Get the table objects.
+  let targetTable = workbook.getTable(TargetTableName);
+  let sourceTable = workbook.getTable(SourceTableName);
+
+  // Check if the tables are there.
+  if (!targetTable || !sourceTable) {
+    console.log(`Required tables missing - Check that both the source (${TargetTableName}) and target (${SourceTableName}) tables are present before running the script.`);
+    return false;
+  }
+
+  return true;
+}
+```
+
+## When to use `throw` the script  
+
+A [`throw` statement](https://developer.mozilla.org/docs/web/javascript/reference/statements/throw) indicates an unexpected error has occurred. It ends the code immediately. For the most part, you don't need to `throw` from your script. Usually, the script automatically informs the user that the script failed to run due to an issue. In most cases, it's sufficient to end the script with an error message and a `return` statement from the `main` function.
+
+However, if your script is running as part of a Power Automate flow, you may want to stop the flow from continuing. A `throw` statement stops the script and tells the flow to stop as well.
+
+The following scripts shows how to use the `throw` statement in our table checking example.
+
+```TypeScript
+function main(workbook: ExcelScript.Workbook) {
+  // These tables must be in the workbook for the script.
+  const TargetTableName = 'Table1';
+  const SourceTableName = 'Table2';
+
+  // Get the table objects.
+  let targetTable = workbook.getTable(TargetTableName);
+  let sourceTable = workbook.getTable(SourceTableName);
+
+  // Check if the tables are there.
+  if (!targetTable || !sourceTable) {
+    // Immediately end the script with an error.
+    throw `Required tables missing - Check that both the source (${TargetTableName}) and target (${SourceTableName}) tables are present before running the script.`;
+  }
+  
+```
+
+## How to use try...catch to handle errors
+
+The [`try...catch`](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Statements/try...catch) technique is a way to detect if an API call failed and handle that error in your script. It may be important to check the return value of an API to verify that it was completed successfully.
+
+Consider the following snippet that performs a large data update on a range.
+
+```TypeScript
+range.setValues(someLargeValues);
+```
+
+If `someLargeValues` is larger than Excel for the web can handle, the `setValues()` call fails. The script then also fails with a [runtime error](../testing/troubleshooting.md#runtime-errors). You may wish to handle this condition in your code. You could either customize the error message (the first of the following two snippets) or break up the update into smaller units (the second snippet). `try...catch` lets you handle this in the script, rather than showing the default error to whoever is using your script.
```

Comment:
```suggestion
If `someLargeValues` is larger than Excel for the web can handle, the `setValues()` call fails. The script then also fails with a [runtime error](../testing/troubleshooting.md#runtime-errors). You may wish to handle this condition in your code. You could either customize the error message or break up the update into smaller units, as shown in the following snippets. `try...catch` lets you handle this in the script, rather than showing the default error to whoever is using your script.
```